core/src/main/java/hudson/cli/DeleteJobCommand.java

```diff
@@ -51,40 +48,17 @@ public String getShortDescription() {
     }
 
     @Override
-    protected int run() throws Exception {
-
-        boolean errorOccurred = false;
-        final Jenkins jenkins = Jenkins.get();
-
-        final HashSet<String> hs = new HashSet<>(jobs);
-
-        for (String job_s: hs) {
-            AbstractItem job;
-
-            try {
-                job = (AbstractItem) jenkins.getItemByFullName(job_s);
-
-                if(job == null) {
-                    throw new IllegalArgumentException("No such job '" + job_s + "'");
-                }
-
-                job.checkPermission(AbstractItem.DELETE);
-                job.delete();
-            } catch (Exception e) {
-                if(hs.size() == 1) {
-                    throw e;
-                }
-
-                final String errorMsg = job_s + ": " + e.getMessage();
-                stderr.println(errorMsg);
-                errorOccurred = true;
-                continue;
-            }
-        }
+    protected void tryDelete(String jobName, Jenkins jenkins) throws Exception{
+        AbstractItem job = (AbstractItem) jenkins.getItemByFullName(jobName);
+        checkExists(job, jobName, "job");
+        job.checkPermission(AbstractItem.DELETE);
+        job.delete();
+        return;
```

Comment:
Unnecessary.

Reply:
Bit lost here. Is it just the checks that are unnecessary? `job.delete();` is the core functionality of this method no?